internal/core/software_renderer.rs

```diff
@@ -1082,14 +1118,13 @@ trait ProcessScene {
     fn process_gradient(&mut self, geometry: PhysicalRect, gradient: GradientCommand);
 }
 
-struct RenderToBuffer<'a, TargetPixel> {
-    buffer: &'a mut [TargetPixel],
-    stride: usize,
+struct RenderToBuffer<'a, TargetPixelBuffer> {
+    buffer: &'a mut TargetPixelBuffer,
     dirty_range_cache: Vec<core::ops::Range<i16>>,
     dirty_region: PhysicalRegion,
 }
 
-impl<T: TargetPixel> RenderToBuffer<'_, T> {
+impl<T: TargetPixel, B: target_pixel_buffer::TargetPixelBuffer<Pixel = T>> RenderToBuffer<'_, B> {
```

Comment:
Do we need T?
Can it be done with

```suggestion
impl<B: target_pixel_buffer::TargetPixelBuffer> RenderToBuffer<'_, B> {
```

And then use `P::Pixel` instead of T

Reply:
Excellent idea. Done :)